src/ssl/flexcon.py

```diff
@@ -31,156 +33,26 @@ def __init__(self, base_estimator, cr=0.05, threshold=0.95, verbose=False):
         self.termination_condition_ = ""
         self.pred_x_it: Dict = {}
         self.cl_memory: List = []
+        self.size_y: int = 0
         self.base_estimator_ = clone(self.base_estimator)
         self.base_estimator_select_ = clone(self.base_estimator)
 
-    def __str__(self):
-        return (
-            f"Classificador {self.base_estimator}\n"
-            f"Outros Parâmetro:"
-            f" CR: {self.cr}\t Threshold: {self.threshold}"
-            f" Máximo IT: {self.max_iter}"
-        )
-
+    @abstractmethod
     def fit(self, X, y):
-        """
-        Fit self-training classifier using `X`, `y` as training data.
-        Parameters
-        ----------
-        X : {array-like, sparse matrix} of shape (n_samples, n_features)
-            Array representing the data.
-        y : {array-like, sparse matrix} of shape (n_samples,)
-            Array representing the labels. Unlabeled samples should have the
-            label -1.
-        Returns
-        -------
-        self : object
-            Fitted estimator.
-        """
-        X, y = self._validate_data(
-            X,
-            y,
-            accept_sparse=["csr", "csc", "lil", "dok"],
-            force_all_finite=False,
-        )
-
-        if y.dtype.kind in ["U", "S"]:
-            raise ValueError(
-                "y has dtype string. If you wish to predict on "
-                "string targets, use dtype object, and use -1"
-                " as the label for unlabeled samples."
-            )
-
-        has_label = y != -1
-        self.cl_memory = [[0] * np.unique(y[has_label]) for _ in range(len(X))]
-
-        if np.all(has_label):
-            warnings.warn("y contains no unlabeled samples", UserWarning)
+        ...
```

Comment:
![image](https://github.com/user-attachments/assets/f80753cc-821b-4521-ad83-3dbb8741b4e3)
![image](https://github.com/user-attachments/assets/c63acd45-e4b6-4eef-a8ec-b0ca83857da7)

```suggestion
        raise NotImplementedError('Implement me')
```

Reply:
feito